pkg/agent/plugin/svidstore/svidstore.go

```diff
@@ -0,0 +1,45 @@
+package svidstore
+
+import (
+	"context"
+	"crypto"
+	"crypto/x509"
+	"time"
+
+	"github.com/spiffe/go-spiffe/v2/spiffeid"
+	"github.com/spiffe/spire/pkg/common/catalog"
+)
+
+type SVIDStore interface {
+	catalog.PluginInfo
+
+	DeleteX509SVID(ctx context.Context, secretsData []string) error
+	PutX509SVID(context.Context, *X509SVID) error
+}
+
+type X509SVID struct {
+	// X509-SVID to be stored
+	SVID *SVID
+
+	// Data relevant for plugin to identify secret
+	SecretsData []string
```

Comment:
Hmm, "secrets" seems a little specific to the target for the first plugin. Generically speaking, isn't this just extra metadata about the SVID to store? Maybe `Metadata []string`?